src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1496,6 +1496,14 @@
 
   {::vacuum-analyze #{"factsets"}})
 
+ (defn add-primary-key-to-resource-events []
+   (jdbc/do-commands
+    ;;As far as we can tell property should never be null but if it is it would
+    ;;cause creating the primary key to fail so we make sure property isn't null
```

Comment:
`property` is nullable to allow the storage of skip events which don't affect any resource property. The current schema definition doesn't make this clear, but the PuppetDB 2.x migration.clj has the following comment:

```
    ;; I wish these next two could be "NOT NULL", but for now we are
    ;; fabricating skipped resources as events, and in those cases we don't
    ;; have any legitimate values to put into these fields.
    ["property" "VARCHAR(40)"]
```

https://github.com/puppetlabs/puppetdb/blob/2.3.8/src/com/puppetlabs/puppetdb/scf/migrate.clj#L265-L268